File: fabric/src/main/java/me/lunaluna/fabric/elytrarecast/Startup.java
package me.lunaluna.fabric.elytrarecast;

import me.lunaluna.fabric.elytrarecast.config.ElytraRecastConfig;
import net.fabricmc.api.ClientModInitializer;

public class Startup implements ClientModInitializer {

	@Override
	public void onInitializeClient() {
		ElytraRecastConfig.load();
	}

}

File: fabric/src/main/java/me/lunaluna/fabric/elytrarecast/Startup.kt
package me.lunaluna.fabric.elytrarecast

import me.lunaluna.fabric.elytrarecast.config.Config
import me.shedaniel.autoconfig.AutoConfig
import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.serializer.GsonConfigSerializer
import net.fabricmc.api.ClientModInitializer

object Startup: ClientModInitializer {
    val config: Config = gsonAutoConfig()
    override fun onInitializeClient() { }
    private inline fun <reified T : ConfigData> gsonAutoConfig(): T = AutoConfig.register(T::class.java) { config, configClass -> GsonConfigSerializer(config, configClass) }.config
}